packages/pyright-internal/src/analyzer/importResolver.ts

```diff
@@ -1197,16 +1204,45 @@ export class ImportResolver {
             }
         }
 
+        if (importType === ImportType.ThirdParty) {
+            const root = this.getParentImportResolutionRoot(filePath, execEnv.root);
+
+            // Go up directories one by one looking for a py.typed file.
+            let current = ensureTrailingDirectorySeparator(getDirectoryPath(filePath));
+            while (this._shouldWalkUp(current, root, execEnv)) {
+                if (this.fileExistsCached(combinePaths(current, 'py.typed'))) {
+                    isThirdPartyPyTypedPresent = true;
```

Comment:
I think you need to check whether it is `py.typed` or `partial py.typed`? and only set it when it is `non partial py.typed`?

Reply:
`getModuleNameInfoFromPath` already walk down from each `import root` to the `file path` given, so, rather than `move` up from `the file path`, we could return `pyTyped` data from `getmoduleNameInfoFromPath` as well along with `moduleName` ?

Reply:
> I think you need to check whether it is py.typed or partial py.typed? and only set it when it is non partial py.typed?

actually, our vfs might already hide `partial py.typed` when it merge them, so, it might not visible to `import resolver`, we probably can test with `bundled stubs` we have in `pylance` since some of them are `partial py.typed`.

Reply:
> it might not visible to import resolver

The return type of `getPyTypedInfo` includes an `isPartiallyTyped` boolean. I can check that. I didn't realize that `isThirdPartyPyTypedPresent` was limited to non-partial `py.typed` files.

> we could return `pyTyped` data from `getmoduleNameInfoFromPath` as well along with `moduleName` ?

Ok, I'll give that a shot.

Reply:
> `getModuleNameInfoFromPath` already walk down from each `import root` to the `file path` give

@heejaechang, where is this happening? I'm not seeing it.

Reply:
ah. sorry, you are right. it just get module name from relative path and not actually walk down.